tools/buildscripts/release.py

```diff
@@ -0,0 +1,338 @@
+
+        if not self.is_dev_release:
+            for key, value in mods_to_disable_in_release.items():
+                pattern = rf'(^{key}\s*:).*?$'
+                content = re.sub(pattern, rf'\1 {value}', content, flags=re.MULTILINE)
+
+        with open(mods_path, mode='w', encoding='utf-8-sig') as file:
+            file.write(content)
+
+    def modify_mods_json(self):
+        mods_path = os.path.join(self.ue4ss_dir, 'Mods/mods.json')
+        with open(mods_path, mode='r', encoding='utf-8-sig') as file:
+            content = json.load(file)
+
+        if self.cpp_mods:
+            for mod_name, mod_info in self.cpp_mods.items():
+                if mod_name not in [mod['mod_name'] for mod in content]:
+                    if not self.is_dev_release:
+                        content.append({'mod_name': mod_name, 'mod_enabled': mod_info['include_in_release']})
+                    else:
+                        content.append({'mod_name': mod_name, 'mod_enabled': True})
+
+        for mod in content:
+            if mod['mod_name'] in self.mods_to_disable_in_release:
+                mod['mod_enabled'] = bool(self.mods_to_disable_in_release[mod['mod_name']])
+
+        with open(mods_path, mode='w', encoding='utf-8-sig') as file:
+            json.dump(content, file, indent=4)
+
+    def copy_executables(self, dwmapi_dll_path, ue4ss_dll_path, ue4ss_pdb_path):
+        shutil.copy(dwmapi_dll_path, self.staging_dir)
+        shutil.copy(ue4ss_dll_path, self.ue4ss_dir)
+        if self.is_dev_release: shutil.copy(ue4ss_pdb_path, self.ue4ss_dir)
+
+    def copy_docs(self):
+        if self.is_dev_release and os.path.exists('docs'):
+            shutil.copytree('docs', os.path.join(self.ue4ss_dir, 'Docs'))
+
+    def modify_changelog(self):
+        changelog_path = self.ue4ss_dir + '/Changelog.md'
+        if os.path.exists(changelog_path): ready_changelog_for_release(changelog_path)
+
+    def package_release(self):
+        version = subprocess.check_output(['git', 'describe', '--tags']).decode('utf-8').strip()
+        main_zip_name = f'zDEV-UE4SS_{version}' if self.is_dev_release else f'UE4SS_{version}'
+        output = os.path.join(self.release_output, main_zip_name)
+        shutil.make_archive(output, 'zip', self.staging_dir)
+        print(f'Created package {output}.zip')
+
+    def cleanup(self):
+        shutil.rmtree(self.staging_dir, ignore_errors=True)
+
+class Packager:
+    """
+    "Release" refers to the zip that is distributed to the end users. It should be in the format UE4SS_<version>.zip and contains only the necessary files.
+    "Dev" refers to the zip that is used for development. It should be in the format zDEV-UE4SS_<version>.zip and contains all the files, pdbs and docs.
+    """
+
+    def __init__(self, args):
+        self.args = args
+        self.release_output = 'release'
+        self.setup()
+
+    def setup(self):
+        shutil.rmtree(self.release_output, ignore_errors=True)
```

Comment:
I'm not sure I agree with ignoring errors. Since this is packaging stuff that is intended for distribution, I think it's better to err on the side of caution when it comes to optional error ignoring. Is it okay to proceed with packaging the project if we don't have a clean release output directory?